backend/corpora/common/dataset_validator.py

```diff
@@ -167,8 +184,75 @@ def verify_uns(self, data_object: anndata.AnnData):
             CorporaConstants.REQUIRED_DATASET_METADATA_FIELDS
             + CorporaConstants.REQUIRED_DATASET_PRESENTATION_METADATA_FIELDS
         ):
-            if metadata_field not in unstructured_metadata_keys:
-                self.log_error_message(metadata_field, "uns", type(data_object).__name__)
+            if metadata_field.field_name not in unstructured_metadata_keys:
+                self.log_error_message(metadata_field.field_name, "uns", type(data_object).__name__)
+            else:
+                self.verify_metadata_type(metadata_field, data_object.uns.get(metadata_field.field_name))
+
+    def verify_metadata_type(self, metadata_property, metadata_values_in_dataset):
+        """
+        Validates the type of each value in a property of an AnnData object.
+
+        Each property value passed in the pandas Series object `metadata_values_in_dataset` is expected to be of the
+        type `metadata_property` where `metadata_property` can either be a type or can be a namedtuple that represents
+        an ontology. When the type is an ontology, the validation instead ensures that each value of the AnnData
+        property belongs to the expected ontology. In some cases, there is an acceptable alternate value not part of
+        the ontology for which the validation also checks.
+        """
+
+        # Canonicalize the metadata values type into a list
+        if isinstance(metadata_values_in_dataset, Series):
+            metadata_values = metadata_values_in_dataset.values
+        else:
+            metadata_values = [metadata_values_in_dataset]
+
+        # Handle the case where the required type is simply a type check.
+        if isinstance(metadata_property.required_type, type):
+            for data_value in metadata_values:
+                if not isinstance(data_value, metadata_property.required_type):
+                    logging.warning(
+                        f"Value {data_value} is not of expected type {metadata_property.required_type} for metadata "
```

Comment:
Maybe say what type `data_value` is? Not a big deal, just helpful for debugging.

Reply:
Done